tests/infra/device_runner.py

```diff
@@ -0,0 +1,112 @@
+# SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Callable, Sequence
+
+import jax
+
+from .device_connector import DeviceType, connector
+from .test_module import TestModule
+
+
+class DeviceRunner:
+    @staticmethod
+    def run_on_tt_device(module: TestModule) -> jax.Array:
+        """Runs test module on TT device."""
+        return DeviceRunner._run_on_device(module, DeviceType.TT)
+
+    @staticmethod
+    def run_on_cpu(module: TestModule) -> jax.Array:
+        """Runs test module on CPU."""
+        return DeviceRunner._run_on_device(module, DeviceType.CPU)
+
+    @staticmethod
+    def run_on_gpu(module: TestModule) -> jax.Array:
+        """Runs test module on GPU."""
+        raise NotImplementedError("Support for GPUs not implemented")
+
+    @staticmethod
+    def put_on_tt_device(*tensors: jax.Array) -> Sequence[jax.Array]:
+        """Puts `tensors` on TT device."""
+        return DeviceRunner._put_on_device(tensors, DeviceType.TT)
+
+    @staticmethod
+    def put_on_cpu(*tensors: jax.Array) -> Sequence[jax.Array]:
+        """Puts `tensors` on CPU."""
+        return DeviceRunner._put_on_device(tensors, DeviceType.CPU)
+
+    @staticmethod
+    def put_on_gpu(*tensors: jax.Array) -> Sequence[jax.Array]:
+        """Puts `tensors` on GPU."""
+        raise NotImplementedError("Support for GPUs not implemented")
+
+    @staticmethod
+    def _run_on_device(
+        module: TestModule,
+        device_type: DeviceType,
+        raise_ex_if_jit_failed: bool = False,
```

Comment:
I think we always want to run jitted graph and to error out if it fails. I guess you wanted to make this method universal so we can use it outside of test module runs, for example in function to create random test inputs. I think the better option than is to create two methods, one which runs jitted graph, and one which doesn't, and then to use the other one explicitly for those specific cases outside of test module runs.

That will also remove the need for this try/except below.